src/args.rs

```diff
@@ -49,17 +64,48 @@ impl Args {
         self.paths.is_empty()
     }
 
-    pub(crate) fn output_mode(&self) -> OutputMode {
-        if self.vimgrep {
-            OutputMode::Vimgrep
+    fn line_number(&self) -> bool {
+        true
+    }
+
+    fn per_match(&self) -> bool {
+        self.vimgrep
```

Comment:
So per above now `--vimgrep` is treated more like how `ripgrep` treats it which is almost like a "shorthand" for a bunch of individual knob settings